diff --git b/godot-codegen/input/gdextension_interface.h a/godot-codegen/input/gdextension_interface.h
index 0b7615f..6db266e 100644
--- b/godot-codegen/input/gdextension_interface.h
+++ a/godot-codegen/input/gdextension_interface.h
@@ -140,22 +140,22 @@ typedef enum {
 
 } GDExtensionVariantOperator;
 
-typedef void *GDExtensionVariantPtr;
-typedef const void *GDExtensionConstVariantPtr;
-typedef void *GDExtensionStringNamePtr;
-typedef const void *GDExtensionConstStringNamePtr;
-typedef void *GDExtensionStringPtr;
-typedef const void *GDExtensionConstStringPtr;
-typedef void *GDExtensionObjectPtr;
-typedef const void *GDExtensionConstObjectPtr;
-typedef void *GDExtensionTypePtr;
-typedef const void *GDExtensionConstTypePtr;
-typedef const void *GDExtensionMethodBindPtr;
+typedef struct TagVariant *GDExtensionVariantPtr;
+typedef const struct TagVariant *GDExtensionConstVariantPtr;
+typedef struct TagStringName *GDExtensionStringNamePtr;
+typedef const struct TagStringName *GDExtensionConstStringNamePtr;
+typedef struct TagString *GDExtensionStringPtr;
+typedef const struct TagString *GDExtensionConstStringPtr;
+typedef struct TagObject *GDExtensionObjectPtr;
+typedef const struct TagObject *GDExtensionConstObjectPtr;
+typedef struct TagType *GDExtensionTypePtr;
+typedef const struct TagType *GDExtensionConstTypePtr;
+typedef struct TagMethodBind *GDExtensionMethodBindPtr;
 typedef int64_t GDExtensionInt;
 typedef uint8_t GDExtensionBool;
 typedef uint64_t GDObjectInstanceID;
-typedef void *GDExtensionRefPtr;
-typedef const void *GDExtensionConstRefPtr;
+typedef struct TagExtensionRef *GDExtensionRefPtr;
+typedef const struct TagExtensionRef *GDExtensionConstRefPtr;
 
 /* VARIANT DATA I/O */
 
